internal/tso/global_allocator.go

```diff
@@ -132,6 +132,9 @@ func (gta *GlobalTSOAllocator) GenerateTSO(count uint32) (uint64, error) {
 			time.Sleep(UpdateTimestampStep)
 			continue
 		}
+		if time.Since(start) > 10*UpdateTimestampStep {
+			log.RatedWarn(1, "slow tso generation", zap.Any("time", time.Since(start)))
```

Comment:
this will print the log every second. Is it too frequent for slow generation?

Reply:
only if the tso response does it display right